CMakeLists.txt

```diff
@@ -290,14 +286,22 @@ endif(NOT NOT_NEED_LIBNSL)
 
 check_function_exists(gethostname HAVE_GETHOSTNAME)
 
+set(OPENSSL_DEFAULT ON)
 if(WIN32)
+  set(OPENSSL_DEFAULT OFF)
   check_library_exists_concat("ws2_32" getch        HAVE_LIBWS2_32)
   check_library_exists_concat("winmm"  getch        HAVE_LIBWINMM)
 endif()
 
-option(CMAKE_USE_OPENSSL "Use OpenSSL code. Experimental" ON)
+option(CMAKE_USE_OPENSSL "Use OpenSSL code. Experimental" ${OPENSSL_DEFAULT})
 mark_as_advanced(CMAKE_USE_OPENSSL)
 
+if(WIN32)
+  CMAKE_DEPENDENT_OPTION(CURL_WINDOWS_SSPI "Use windows libraries to allow NTLM authentication without openssl" OFF
+    "CMAKE_USE_OPENSSL" ON)
```

Comment:
IIUC the goal is to present the `CURL_WINDOWS_SSPI` option (with default `ON`) when `CMAKE_USE_OPENSSL` is `OFF` (default).  When `CMAKE_USE_OPENSSL` is `ON` then `CURL_WINDOWS_SSPI` should be non-optional and forced to `OFF`.  In that case the dependent option should be:

      CMAKE_DEPENDENT_OPTION(CURL_WINDOWS_SSPI "Use windows libraries to allow NTLM authentication without openssl" ON
        "NOT CMAKE_USE_OPENSSL" OFF)